wwwapp/models.py

```diff
@@ -85,33 +126,15 @@ class UserInfo(models.Model):
     class Meta:
         permissions = (('see_user_info', 'Can see user info'),)
 
-    """
-    Retrieves the birth date from the provided PESEL number.
-    Returns a date class instance or None if the PESEL is malformed or not present.
-    """
     def get_birth_date(self) -> date or None:
-        birth = self.pesel[:6]
-        if birth is not None and birth.isdigit() and len(birth) == 6:
-            year = int(birth[:2], 10)
-            month = int(birth[2:4], 10)
-            days = int(birth[4:6], 10)
-
-            if month < 20:
-                year += 1900
-            elif month < 40:
-                month -= 20
-                year += 2000
-            elif month < 60:
-                month -= 40
-                year += 2100
-            elif month < 80:
-                month -= 60
-                year += 2200
-            elif month < 100:
-                month -= 80
-                year += 1800
-
-            return date(year, month, days)
+        """
+        Retrieves the birth date from the provided PESEL number.
+        Returns a date class instance or None if the PESEL is malformed or not present.
+        """
+        if not self.pesel or len(self.pesel) < 6:
+            return None
+        return PESELField._extract_date(self.pesel)
```

Comment:
Why not `self.pesel.extract_date()`?

Reply:
pesel is a string in this context.

Reply:
Ahh, right. In that case, maybe it makes more sense to use CharField with a validator rather than a custom field?

Reply:
@krzys-h I don't have a preference here.

Reply:
I think we should leave this as is. Maybe somebody will create a custom widget for PESEL Field in the future.